speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/DecisionMaker.inl

```diff
@@ -102,16 +103,18 @@ bool TEMP_DECISIONMAKER::Decide(tCarElt* p_car, tSituation* p_situation, unsigne
 
     if (m_recorder)
     {
-        const float decisionValues[DECISION_RECORD_PARAM_AMOUNT] = {m_decision.GetBrake(), m_decision.GetSteer()};
-        m_recorder->WriteDecisions(decisionValues, p_tickCount);
+        float* decisionsVals = new float[DECISION_RECORD_PARAM_AMOUNT];
+        decisionsVals[0] = m_decision.GetBrake();
+        decisionsVals[1] = m_decision.GetSteer();
+        m_recorder->WriteDecisions(decisionsVals, p_tickCount);
```

Comment:
This causes a memory leak since it's never deleted. But keeping it stack allocated would be preferable. You can then copy the values in your mock instead of saving a reference to this variable.

Reply:
I agree with this, the size of the array is constant and thus should be allocated on the stack

Reply:
I now decided to pass m_decision as a whole. The conversion to a stack allocated float array happens in the specific recordere itself.